tests-e2e/core/gitopsDeployment_test.go

```diff
@@ -0,0 +1,534 @@
+		k8sClient, err := fixture.GetKubeClient()
+
+		Expect(err).To(BeNil())
+		ctx := context.Background()
+
+		It("Should ensure succesfull creation of GitOpsDeployment", func() {
+
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			By("creating the GitOpsDeployment")
+
+			gitOpsDeploymentResource := buildGitOpsDeploymentResource(name,
+				repoURL, "environments/overlays/dev",
+				managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated)
+
+			err := k8s.Create(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			Eventually(gitOpsDeploymentResource, "2m", "1s").Should(
+				SatisfyAll(
+					gitopsDeplFixture.HaveSyncStatusCode(managedgitopsv1alpha1.SyncStatusCodeSynced),
+					gitopsDeplFixture.HaveHealthStatusCode(managedgitopsv1alpha1.HeathStatusCodeHealthy),
+				),
+			)
+			err = k8s.Delete(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+			err = k8s.Get(&gitOpsDeploymentResource)
+			Expect(err).ToNot(Succeed())
+
+		})
+
+		It("Should ensure synchronicity of create and update of GitOpsDeployment", func() {
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			By("ensurng no updates are done in existing deployment, if CR is submitted again without any changes")
+
+			gitOpsDeploymentResource := buildGitOpsDeploymentResource(name,
+				repoURL, "environments/overlays/dev",
+				managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated)
+
+			err := k8s.Create(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			gitOpsDeploymentResource.Spec.Source.RepoURL = repoURL
+			gitOpsDeploymentResource.Spec.Source.Path = "environments/overlays/dev"
+
+			err = k8s.Update(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&gitOpsDeploymentResource), &gitOpsDeploymentResource)
+			Expect(gitOpsDeploymentResource.Spec.Source.RepoURL).To(Equal(repoURL))
+			Expect(gitOpsDeploymentResource.Spec.Source.Path).To(Equal("environments/overlays/dev"))
+
+			Eventually(gitOpsDeploymentResource, "2m", "1s").Should(
+				SatisfyAll(
+					gitopsDeplFixture.HaveSyncStatusCode(managedgitopsv1alpha1.SyncStatusCodeSynced),
+					gitopsDeplFixture.HaveHealthStatusCode(managedgitopsv1alpha1.HeathStatusCodeHealthy),
+				),
+			)
+
+			err = k8s.Delete(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+			err = k8s.Get(&gitOpsDeploymentResource)
+			Expect(err).ToNot(Succeed())
+		})
+
+		It("Should ensure synchronicity of create and update of GitOpsDeployment", func() {
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+			By("ensuring GitOpsDeployment should update successfully on changing value(s) within Spec")
+
+			gitOpsDeploymentResource := buildGitOpsDeploymentResource(name,
+				"https://github.com/redhat-appstudio/managed-gitops", "manifests/appstudio-controller-rbac",
+				managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated)
+
+			err := k8s.Create(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
```

Comment:
After create, we need to wait for Argo CD to deploy the resource (e.g. an `Eventually` call). 

One thing to note: just because a K8s resource has been created, it does not mean that the particular action that a resource describes has completed.

For example, in this case, just because we have created a `GitOpsDeployment`, e.g. just because the `Create` call has completed, it does not mean the the Git repository has been deployed to the cluster.

Calling `k8s.Create(...)` only _starts_ the process, it does not indicate that it is complete. (e.g. it is an asynchronous operation)

So, as mentioned, here we should wait for Synched/Healthy, and it also couldn't hurt to wait for (`Eventually`) at least one K8s resource from the Git repository was deployed.